src/index.ml

```diff
@@ -236,33 +250,38 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
         Log.debug (fun l ->
             l "[%s] v fresh=%b readonly=%b" (Filename.basename root) fresh
               readonly);
-        let t = v ~fresh ~readonly ~log_size root in
-        Hashtbl.add roots (root, readonly) t;
-        t
+        let instance = v ~fresh ~readonly ~log_size root in
+        Hashtbl.add roots (root, readonly) instance;
+        ref (Some instance)
     in
     `Staged f
 
   let v_no_cache ~fresh ~readonly ~log_size root =
     let lock =
       if not readonly then Some (IO.lock (lock_path root)) else None
     in
-    let config = { log_size = log_size * entry_size; readonly } in
+    let config = { log_size = log_size * entry_size; readonly; fresh } in
     let log_path = log_path root in
-    let index_path = index_path root in
-    let log_mem = Tbl.create 1024 in
-    let log = IO.v ~fresh ~readonly ~generation:0L ~fan_size:0L log_path in
-    let generation = IO.get_generation log in
+    let log =
+      if readonly && not (Sys.file_exists log_path) then None
```

Comment:
Does this lead to weirdness if a read/write instance is opened _after_ a read-only instance has been initialised? It seems that the read-only instance won't see the log.

Reply:
Hmm, no, scratch that. I see `try_load_log` now :slightly_smiling_face: